rust/src/builder.rs

```diff
@@ -234,6 +232,9 @@ pub trait ArrayBuilder {
     /// type. In this case, one can first cast this into a `Any`, and then use
     /// `downcast_mut` to get a reference on the specific builder.
     fn as_any_mut(&mut self) -> &mut Any;
+
+    /// Returns the boxed builder as a box of `Any`.
+    fn to_any(self: Box<Self>) -> Box<Any>;
```

Comment:
I think it's convention for `to_..` methods to take self by reference

Reply:
Found [reference](https://github.com/rust-lang/rust-clippy/wiki#wrong_self_convention)

Reply:
Changed to `into_box_any`. Let me know if this is better.